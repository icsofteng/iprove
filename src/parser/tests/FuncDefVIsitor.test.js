import {parse} from '../'

test("Visitor Test Func Def friends", ()=> {
    expect(parse("define friends(Person, Person):Bool")).toMatchSnapshot()
  })

test("Visitor Test Func Def Dragon", ()=> {
    expect(parse("define Power(dragon, Dragon):int")).toMatchSnapshot()
})

test("Visitor Test Func Def Happy", ()=> {
    expect(parse("define Happy(Dragon, Human):reward")).toMatchSnapshot()
})

