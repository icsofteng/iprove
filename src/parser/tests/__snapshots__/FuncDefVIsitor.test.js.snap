// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visitor Test Func Def Dragon 1`] = `
Object {
  "ast": Array [
    Object {
      "name": "Power",
      "params": Array [
        Object {
          "type": "type",
          "value": "Dragon",
        },
        Object {
          "type": "type",
          "value": "Dragon",
        },
      ],
      "returnType": Object {
        "type": "type",
        "value": "Int",
      },
      "type": "funcDef",
    },
  ],
  "errors": false,
  "functions": Array [
    Object {
      "name": "Power",
      "params": Array [
        "Dragon",
        "Dragon",
      ],
      "rType": "Int",
    },
  ],
  "identifiers": Array [],
  "relations": Array [],
  "types": Array [
    "Dragon",
  ],
}
`;

exports[`Visitor Test Func Def Happy 1`] = `
Object {
  "ast": Array [
    Object {
      "name": "Happy",
      "params": Array [
        Object {
          "type": "type",
          "value": "Dragon",
        },
        Object {
          "type": "type",
          "value": "Human",
        },
      ],
      "returnType": Object {
        "type": "type",
        "value": "Reward",
      },
      "type": "funcDef",
    },
  ],
  "errors": false,
  "functions": Array [
    Object {
      "name": "Happy",
      "params": Array [
        "Dragon",
        "Human",
      ],
      "rType": "Reward",
    },
  ],
  "identifiers": Array [],
  "relations": Array [],
  "types": Array [
    "Reward",
    "Dragon",
    "Human",
  ],
}
`;

exports[`Visitor Test Func Def friends 1`] = `
Object {
  "ast": Array [
    Object {
      "name": "friends",
      "params": Array [
        Object {
          "type": "type",
          "value": "Person",
        },
        Object {
          "type": "type",
          "value": "Person",
        },
      ],
      "returnType": Object {
        "type": "type",
        "value": "Bool",
      },
      "type": "funcDef",
    },
  ],
  "errors": false,
  "functions": Array [
    Object {
      "name": "friends",
      "params": Array [
        "Person",
        "Person",
      ],
      "rType": "Bool",
    },
  ],
  "identifiers": Array [],
  "relations": Array [],
  "types": Array [
    "Person",
  ],
}
`;
